src/core/service.c

```diff
@@ -947,6 +947,24 @@ static void service_search_main_pid(Service *s) {
                 log_unit_warning_errno(UNIT(s), r, "Failed to watch PID "PID_FMT" from: %m", pid);
 }
 
+bool service_exited_release_console(Unit *u)
+{
+        Service *s;
+        if (u->type != UNIT_SERVICE)
+            return false;
+
+        s = SERVICE(u);
```

Comment:
The SERVICE macro already includes the check if the type is UNIT_SERVICE, so you should skip the type check above, and instead check that `s` is not NULL.